packages/lesswrong/components/tagging/TagPage.tsx

```diff
@@ -1080,25 +1104,54 @@ const TagPage = () => {
 
   const currentUser = useCurrentUser();
   const { query, params: { slug } } = useLocation();
+  const { lens: lensSlug } = query;
   // const { onOpenEditor } = useContext(TagEditorContext);
   
   // Support URLs with ?version=1.2.3 or with ?revision=1.2.3 (we were previously inconsistent, ?version is now preferred)
   const { version: queryVersion, revision: queryRevision } = query;
   const revision = queryVersion ?? queryRevision ?? null;
 
   const contributorsLimit = 16;
-  const tagQueryOptions: Partial<UseMultiOptions<"TagPageWithArbitalContentFragment" | "TagPageRevisionWithArbitalContentFragment", "Tags">> = {
+
+  let tagFragmentName: FragmentTypesByCollection['Tags'] = "TagPageWithArbitalContentFragment";
+  let tagQueryOptions: Partial<UseMultiOptions<"TagPageWithArbitalContentFragment" | "TagPageRevisionWithArbitalContentFragment" | "TagPageWithArbitalContentAndLensRevisionFragment", "Tags">> = {
     extraVariables: {
-      ...(revision ? { version: 'String' } : {}),
       contributorsLimit: 'Int',
     },
     extraVariablesValues: {
-      ...(revision ? { version: revision } : {}),
       contributorsLimit,
     },
   };
 
-  const tagFragmentName = revision ? "TagPageRevisionWithArbitalContentFragment" : "TagPageWithArbitalContentFragment";
+  if (revision && !lensSlug) {
+    tagFragmentName = "TagPageRevisionWithArbitalContentFragment";
+    tagQueryOptions = {
+      extraVariables: {
+        version: 'String',
+        contributorsLimit: 'Int',
+      },
+      extraVariablesValues: {
+        version: revision,
+        contributorsLimit,
+      },
+    };
+  }
+
+  if (revision && lensSlug) {
+    tagFragmentName = "TagPageWithArbitalContentAndLensRevisionFragment";
+    tagQueryOptions = {
+      extraVariables: {
+        version: 'String',
+        lensSlug: 'String',
+        contributorsLimit: 'Int',
+      },
+      extraVariablesValues: {
+        version: revision,
+        lensSlug,
+        contributorsLimit,
+      },
+    };
+  }
```

Comment:
I think it's probably time to extract assignments for these two variables into a function